determined-ai/determined
# fix: Count only active tasks in cluster info board

## Description

Fixing a bug for the release - previously any TensorBoard / JupyterLab / Command / Shell which would be visible on the Tasks page was being counted, but we really only want the task.state == Active ones.

Alternatives which didn't work:
- the GetTensorboardsRequest and other request objects can be filtered by user, but not by state
- I was unable to write a count function that accepts all types. The proto objects cannot be made subclasses of a universal Task type. I also tried converting the response's `[]*tensorboardv1.Tensorboard` into a new `[]*universalTask` type requiring only a State, but type conversion might require looping over each element in the slice anyway.

## Test Plan

- Start a TensorBoard or JupyterLab.
- Observe that it is counted as Active on the Cluster page.
- Kill the task on the Tasks page. Observe that it is still listed as Terminated on the Tasks page.
- Return to the Cluster page; it is not counted.


## Checklist

See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
